app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesStore.kt

```diff
@@ -0,0 +1,32 @@
+package de.tum.`in`.tumcampusapp.component.tumui.grades
+
+import android.content.SharedPreferences
+import javax.inject.Inject
+
+data class GradesStoreState(
+        val isFirstRefresh: Boolean,
+        val existingGrades: List<String>
+)
+
+class GradesStore @Inject constructor(
+        private val sharedPrefs: SharedPreferences
+) {
+
+    private val isFirstRefresh: Boolean
+        get() = sharedPrefs.contains(KEY_GRADED_COURSES).not()
```

Comment:
Yea... no. This is not the best idea. Besides polluting the shared prefs with stuff that should not be in there, it is a rather cumbersome solution for the initialization problem. How about just skipping the notification, if there are more than 5 new grades? Seems a lot simpler and in practice, nobody will receive 5 grades at exactly the same time or within a refresh period. 

Thoughts?

Reply:
(Yes, magic constants and so on, that @pfent will complain about, but its a simpler solution for that problem)

Reply:
Yes, I was not too happy with the initialization problem either. Will update as per your suggestion.